pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2553,6 +2553,13 @@ impl Timeline {
                 .open_mut()?
                 .finish_gc_compaction(&layer_selection, &compact_to, &self.metrics)
         };
+
+        // Schedule an index-only upload to update the `latest_gc_cutoff` in the index_part.json.
+        // Otherwise, after restart, the index_part only contains the old `latest_gc_cutoff` and
+        // find_gc_cutoffs will try accessing things below the cutoff. TODO: ideally, this should
+        // be batched into `schedule_compaction_update`.
+        let disk_consistent_lsn = self.disk_consistent_lsn.load();
+        self.schedule_uploads(disk_consistent_lsn, None)?;
```

Comment:
do we need a barrier here?

Reply:
No, we only need to put them into the queue. As long as the `schedule_compaction_update` happens after `schedule_uploads`, the remote view of the timeline would be consistent.